bittensor/commands/stake.py

```diff
@@ -566,3 +568,348 @@ def add_args(parser: argparse.ArgumentParser):
 
         bittensor.wallet.add_args(list_parser)
         bittensor.subtensor.add_args(list_parser)
+
+
+class SetChildCommand:
+    """
+    Executes the ``set_child`` command to add a child hotkey on a specified subnet on the Bittensor network.
+
+    This command is used to delegate authority to different hotkeys, securing their position and influence on the subnet.
+
+    Usage:
+        Users can specify the amount or 'proportion' to delegate to a child hotkey (either by name or ``SS58`` address),
+        the user needs to have sufficient authority to make this call, and the sum of proportions cannot be greater than 1.
+
+    The command prompts for confirmation before executing the set_child operation.
+
+    Example usage::
+
+        btcli stake set_child --child <child_hotkey> --hotkey <parent_hotkey> --netuid 1 --proportion 0.3
+
+    Note:
+        This command is critical for users who wish to delegate child hotkeys among different neurons (hotkeys) on the network.
+        It allows for a strategic allocation of authority to enhance network participation and influence.
+    """
+
+    @staticmethod
+    def run(cli: "bittensor.cli"):
+        r"""Set child hotkey."""
+        try:
+            subtensor: "bittensor.subtensor" = bittensor.subtensor(
+                config=cli.config, log_verbose=False
+            )
+            SetChildCommand._run(cli, subtensor)
+        finally:
+            if "subtensor" in locals():
+                subtensor.close()
+                bittensor.logging.debug("closing subtensor connection")
+
+    @staticmethod
+    def _run(cli: "bittensor.cli", subtensor: "bittensor.subtensor"):
+        wallet = bittensor.wallet(config=cli.config)
+
+        current_proportions = GetChildrenCommand.run(cli)
+
+        # Get values if not set.
+        if not cli.config.is_set("netuid"):
+            cli.config.netuid = int(Prompt.ask("Enter netuid"))
+
+        if not cli.config.is_set("child"):
+            cli.config.child = Prompt.ask("Enter child hotkey (ss58)")
+
+        if not cli.config.is_set("hotkey"):
+            cli.config.hotkey = Prompt.ask("Enter parent hotkey (ss58)")
+
+        if not cli.config.is_set("proportion"):
+            cli.config.proportion = Prompt.ask("Enter proportion")
+
+        # Parse from strings
+        netuid = cli.config.netuid
+
+        try:
+            proportion = float(cli.config.proportion)
+        except ValueError:
+            console.print(
+                ":cross_mark:[red] Invalid proportion amount[/red] [bold white]{}[/bold white]".format(
+                    cli.config.proportion
+                )
+            )
+            sys.exit()
+            
+        total_proposed = proportion + current_proportions
+        if total_proposed > 1:
+            raise ValueError(f":cross_mark:[red] The sum of all proportions cannot be greater than 1. Proposed sum of proportions is {total_proposed}[/red]")
```

Comment:
What do you think about moving this to extrinsics? That way users can call it through the subtensor and still get this validation